### ISSUE TITLE
cqn2sql issue with CROSS joins

### ISSUE THREAD

**Issue by beleczkia** [cap-js/cds-dbs#897]
### Description of erroneous behaviour 

We have a query for filtering which uses cross join produced by cds.parse.cql function. During execution we run into errors caused by `undefined` where expression.

After some investigation it turned out that cqn2sql.js tries to create ON clause with a `from.on` which is missing for the `CROSS JOIN`.

```js
  if (from.join)
        return `${this.from(from.args[0])} ${from.join} JOIN ${this.from(from.args[1])} ON ${this.where(from.on)}`
```

This issue is reproducible in `cds repl`.

### Detailed steps to reproduce

> 1. cds repl 
> 2. const {SQLService} = require('@cap-js/db-service')
> 3. query = SELECT.from('foo').where(['exists', cds.parse.cql('select * from bar1, bar2')])
> 4. query.toSQL()

### Details about your project

| Your Project Name | https://github.com/your/repo |
|:------------------|---------------------------------------|
| @cap-js/asyncapi | 1.0.2 |
| @cap-js/openapi | 1.0.7 |
| @sap/cds | 8.4.1 |
| @sap/cds-compiler | 5.4.2 |
| @sap/cds-dk | 8.4.1 |
| @sap/cds-fiori | 1.2.7 |
| @sap/cds-foss | 5.0.1 |
| @sap/cds-mtxs | 2.3.0 |
| @sap/eslint-plugin-cds | 3.1.0 |
| Node.js | v20.17.0 |

**Comment by patricebender** [comment]
what is the error? Is it the same as reported in #891

**Comment by beleczkia** [comment]
Yes.

**Comment by patricebender** [comment]
fixed with [`@cap-js/db-service v1.15.1`](https://www.npmjs.com/package/@cap-js/db-service)
